homeassistant/components/minecraft_server/api.py

```diff
@@ -0,0 +1,137 @@
+"""API for the Minecraft Server integration."""
+
+
+from dataclasses import dataclass
+from enum import StrEnum
+import logging
+
+from mcstatus import BedrockServer, JavaServer
+from mcstatus.status_response import BedrockStatusResponse, JavaStatusResponse
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class MinecraftServerData:
+    """Representation of Minecraft Server data."""
+
+    # Common data
+    latency: float
+    motd: str
+    players_max: int
+    players_online: int
+    protocol_version: int
+    version: str
+
+    # Data available only in 'Java Edition'
+    players_list: list[str] | None
+
+    # Data available only in 'Bedrock Edition'
+    edition: str | None
+    game_mode: str | None
+    map_name: str | None
+
+
+class MinecraftServerType(StrEnum):
+    """Enumeration of Minecraft Server types."""
+
+    JAVA_EDITION = "Java Edition"
+    BEDROCK_EDITION = "Bedrock Edition"
```

Comment:
```suggestion
    BEDROCK_EDITION = "Bedrock Edition"
    JAVA_EDITION = "Java Edition"
```

swap around them to use a loop in the config flow